transformations/negate_strengthen/transformation.py

```diff
@@ -0,0 +1,462 @@
+
+
+    def text_to_negated_edits(self, _text, sentid2tid, get_roots=None):
+        # clean multiple whitespaces
+        _text = _RE_COMBINE_WHITESPACE.sub(" ", _text).strip()
+        text = _text.split(' ')
+        pos = [p[1] for p in pos_tag(text)]
+
+        # get roots based on dep info from spacy
+        table = self.text_to_spacy_table_info(_text)
+        if get_roots is None:
+            get_roots = table[([True if t[0:2] =='VB' else False for t in table['pos_tag']]) & (table['dep'] == 'ROOT')].index
+        if self.verbose:
+            print(">>>>> get_roots: ", get_roots)
+        itemdict = {'text': [(int(idx+1),0,int(idx),t) for idx, t in enumerate(text)]}
+
+        t_dict = {}
+        for jx, root in enumerate(get_roots):
+            # check if root has acomp, adjust root
+            if table.loc[root,'lemma'] in MODAL_TYPE_LEMMA or (table.loc[root,'pos_tag'] in ['MD']):
+                if len(table[(table['dep'].isin(['acomp','attr'])) & (table['head_id']==root)])>0:
+                    root = table[(table['dep'].isin(['acomp','attr'])) & (table['head_id']==root)].index[0]
+                    if self.verbose:
+                        print('new root: {}'.format(table.loc[root, 'text']))
+            
+            # using root loc, negate
+            edit_text, method, edit_id = self.negation_rules(
+                root+1, itemdict, text, pos,
+                sentid2tid, method=[], num_tries=2)
+
+            if self.verbose:
+                print(">>>>> edit_text: ", edit_text)
+
+            # improve text flow
+            if None not in method and len(method)>0:
+                alt_text, alt_word = self.improve_negation_flow(_text.split(' '), edit_id, method)
+                # if method is nor format
+                if alt_text is not None:
+                    if method[-1]=='JJ_1_2':
+                        alt_text, alt_word_2 = self.improve_negation_flow(alt_text.copy(), edit_id+2, method)
+                        alt_word = [alt_word, alt_word_2]
+            else:
+                alt_word = None
+                alt_text = None
+
+            # keep only successful edits
+            if None not in method:
+                t_dict[jx] = {
+                    'edit_text': edit_text,
+                    'edit_id': edit_id,
+                    'edit_method': method,
+                    'alt_word': alt_word,
+                    'alt_text': alt_text
+                }
+
+        return t_dict
+
+
+    def text_to_stronger_edits(self, _text, sentid2tid):
+        # clean multiple whitespaces
+        _text = _RE_COMBINE_WHITESPACE.sub(" ", _text).strip()
+        text = _text.split(' ')
+
+        # get weaker words for converting
+        table = self.text_to_spacy_table_info(_text)
+        get_roots = table[table['lemma'].isin(MODAL_STRENGTHEN_DICT.keys())].index
+        itemdict = {'text': [(int(idx+1),0,int(idx),t) for idx, t in enumerate(text)]}
```

Comment:
Can we remove 'itemdict' if we are not using it?

Reply:
Ok, done!